### ISSUE TITLE
docker_image module: "Either path or fileobj needs to be provided"

### ISSUE THREAD

**Issue by vikas027** [ansible/ansible-modules-core#2991]
##### Issue Type

Bug Report
##### Component Name

docker_image module
##### Ansible Version

2.0.0.2
##### Environment

OSX: 10.11.2
Ansible Host: 2.0.0.2
Python: 2.7.10
docker-py: 1.7.0
requests: 2.9.1
Ansible Remote Client: Amazon Linux
##### Problem

I was just want to use [docker_image](http://docs.ansible.com/ansible/docker_image_module.html) module to pull an image.  

``` yaml
- docker_image: name='vikas027/docker-redis:latest' state=present
```
##### Error

Documentation says that `path` is not mandatory, but I got the below error. Am I missing something ?

``` bash
failed: [myserver] => (item=vikas027/docker-redis:latest) => {"failed": true, "item": "vikas027/docker-redis:latest", "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/ec2-user/.ansible/tmp/ansible-tmp-1454933036.64-112508914115589/docker_image\", line 2391, in <module>\r\n    main()\r\n  File \"/home/ec2-user/.ansible/tmp/ansible-tmp-1454933036.64-112508914115589/docker_image\", line 401, in main\r\n    image_id = manager.build()\r\n  File \"/home/ec2-user/.ansible/tmp/ansible-tmp-1454933036.64-112508914115589/docker_image\", line 293, in build\r\n    stream = self.client.build(self.path, dockerfile=self.dockerfile, tag=':'.join([self.name, self.tag]), nocache=self.nocache, rm=True, stream=True)\r\n  File \"/usr/local/lib/python2.7/site-packages/docker/api/build.py\", line 24, in build\r\n    raise TypeError(\"Either path or fileobj needs to be provided.\")\r\nTypeError: Either path or fileobj needs to be provided.\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

**Comment by ebel** [comment]
Looking at [docker_image.py](https://github.com/ansible/ansible-modules-core/blob/e318be30ff0a4c19e61ff15437ff96a6ba57f696/cloud/docker/docker_image.pyl)

It looks like we are expecting a 'path' value on both lines 195 and 
292.  Looks to me that it should be required.

I will play around with, and see what I find.

**Comment by ndemoor** [comment]
+1

**Comment by hatt** [comment]
It seems to me more like the path should be made properly optional? Use case being fetching an image but not running it or building it to bootstrap VM images.

**Comment by Smana** [comment]
+1

**Comment by bartmeuris** [comment]
+1

**Comment by relet** [comment]
+1

**Comment by conradkleinespel** [comment]
Seems like just adding `path: .` fixes this for me, for anyone looking for a quick workaround.

**Comment by sszwed** [comment]
+1

**Comment by grossws** [comment]
Workaround by @conradkleinespel doesn't work for me (ansible 2.0.2.0):

``` json
{
  "changed": true, 
  "failed": true, 
  "item": {"name": "grossws/centos", "tag": 7}, 
  "msg": "Docker API error: Cannot locate specified Dockerfile: Dockerfile"
}
```

**Comment by dmitrybelyakov** [comment]
Yeah, same here...

**Comment by jctanner** [comment]
bot_broken

**Comment by chouseknecht** [comment]
bug_resolved

I verified this is fixed in Ansible 2.1.0

Given the following playbook that executes the task reported by @vikas027:

```
---
- hosts: localhost
  connection: local
  gather_facts: no
  tasks:
    - docker_image: name='vikas027/docker-redis:latest' state=present
```

The playbook runs and pulls the image. No _path_ required.

The only time a path is required is when a build is to be performed. To perform a build provide a path to the image context.
